ls1intum/Artemis
# Refactor exam file upload exercises and fix bug with summary

### Checklist

### Motivation and Context


We refactored the exam-file-upload exercise. 

### Description

It now is more independent of the exam-participation.component, and does all the saving on its own. We also fixed f issue where the `submission.submitted` was set to `false` when the exam was refreshed. This lead to a problem because file-upload exercises did not set it to the correct `true` value when handing in, so the submission was shown as not handed in, in the summary.

### Steps for Testing


1. Create an exam with multiple exercises, among them at least one file-upload exercise.
2. First test with a test run: create one, and start. Submit some of the exercises, also the file-upload exercise. (and some not)
3. Now refresh the exam, after going through the cover all previously submitted submissions should still be submitted. (the blue check mark in the navigation bar)
4. Hand in the exam, check if everything is correct on the summary page
5. Repeat steps with a standard exam

### Test Coverage
